go/host/storage/storage.go

```diff
@@ -34,7 +35,10 @@ func (s *storageImpl) AddBatch(batch *common.ExtBatch) error {
 	}
 
 	if err := hostdb.AddBatch(dbtx, s.db.GetSQLStatement(), batch); err != nil {
-		if err := dbtx.Rollback(); err != nil {
+		if err1 := dbtx.Rollback(); err1 != nil {
+			return err1
+		}
+		if errors.Is(err, errutil.ErrAlreadyExists) {
```

Comment:
this returns the "ErrAlreadyExists", which is part of the logical API of this method.
So that the client can handle when a batch is inserted again